pkg/network/network.go

```diff
@@ -29,22 +31,34 @@ The hostname resolves to the following ips:
 
 // ValidateHostnameUniqueness Validate that the given hostname resolves to at most 1 ip per ip version.
 func ValidateHostnameUniqueness(addr string) error {
-	resolver := net.DefaultResolver
+	var errs error
+	errCount := 0
 	ctx := context.Background()
-
-	// errors ignored since we don't know if
-	// both v4 and v6 is supported.
-	// for example,  *net.AddrError can return
-	// - no suitable address found
-	// if ipv6 is not supported on the host.
-	ipv4, _ := resolver.LookupIP(ctx, "ip4", addr)
-	ipv6, _ := resolver.LookupIP(ctx, "ip6", addr)
-
-	if len(ipv4) > 1 || len(ipv6) > 1 {
+	resolver := net.DefaultResolver
+	ipv4, err := resolver.LookupIP(ctx, "ip4", addr)
+	if err != nil {
+		errCount++
+		err = fmt.Errorf("ipv4: %w", err)
+		errs = multierror.Append(err, errs)
```

Comment:
```suggestion
		errs = multierror.Append(errs, fmt.Errorf("ipv4: %w", err))
```
looks like you've mixed up the order of Append,
See https://github.com/hashicorp/go-multierror#usage for example